arraykit.c

```diff
@@ -132,6 +137,108 @@ name_filter(PyObject *Py_UNUSED(m), PyObject *n)
     return n;
 }
 
+
+static PyObject *
+shape_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // Represent a 1D array as a 2D array with length as rows of a single-column array.
+    // https://numpy.org/doc/stable/reference/c-api/array.html
+    // https://stackoverflow.com/questions/56182259/how-does-one-acces-numpy-multidimensionnal-array-in-c-extensions
+
+    PyObject *shape = PyTuple_New(2);
+    if (!shape) {
+        return NULL;
+    }
+
+    PyArrayObject *array = (PyArrayObject *)a;
+
+    // get size at dimension 0, set in tuple position 0
+    int size0 = PyArray_DIM(array, 0);
+    PyObject *rows = PyLong_FromLong(size0);
+    if (!rows) {
+        Py_DECREF(shape);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(shape, 0, rows);
+
+    // handle 1 dimensional array
+    if (PyArray_NDIM(array) == 1) {
+        PyObject *cols = PyLong_FromLong(1);
+        if (!cols) {
+            Py_DECREF(shape);
+            return NULL;
+        }
+        PyTuple_SET_ITEM(shape, 1, cols);
+        return shape;
+    }
+    // get size at dimension 1, set in tuple position 1
+    int size1 = PyArray_DIM(array, 1);
+    PyObject *cols = PyLong_FromLong(size1);
+    if (!cols) {
+        Py_DECREF(shape);
+        return NULL;
+    }
+    PyTuple_SET_ITEM(shape, 1, cols);
+    return shape;
+}
+
+static PyObject *
+column_2d_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // Reshape a flat ndim 1 array into a 2D array with one columns and rows of length.
+    // related example: https://github.com/RhysU/ar/blob/master/ar-python.cpp
+    PyArrayObject *array = (PyArrayObject *)a;
+
+    if (PyArray_NDIM(array) == 1) {
+        // https://numpy.org/doc/stable/reference/c-api/types-and-structures.html#c.PyArray_Dims
+        npy_intp dim[2] = {PyArray_DIM(array, 0), 1};
+        PyArray_Dims shape = {dim, 2};
+        // PyArray_Newshape might return NULL and set PyErr, so no handling to do here
+        return PyArray_Newshape(array, &shape, NPY_ANYORDER); // already a PyObject*
+    }
+    Py_INCREF(a); // returning borrowed ref, must increment
+    return a;
+}
+
+static PyObject *
+column_1d_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // Ensure that a column that might be 2D or 1D is returned as a 1D array.
+    PyArrayObject *array = (PyArrayObject *)a;
+
+    if (PyArray_NDIM(array) == 2) {
+        npy_intp dim[1] = {PyArray_DIM(array, 0),};
```

Comment:
```suggestion
        npy_intp dim[1] = {PyArray_DIM(array, 0)};
```

Reply:
Nice catch. This is me thinking a struct literal with one element is like a tuple with one element, requiring an extra comma!